### PR TITLE
DAOS-9558 control: fix storage usage statistic

### PR DESCRIPTION
**Pull request by knard-intel**
Draft PR => DO NOT MERGE AS IT

This patch fix the calculation of the storage usage statistic return by a query on storage usage.
Only, manual tests have been done.

Signed-off-by: Cedric Koch-Hofer <cedric.koch-hofer@intel.com>

### CODE REVIEW

File: src/control/server/ctl_storage_rpc.go [hunk 1]
```diff
@@ -151,6 +159,112 @@ func (c *ControlService) scanScm(ctx context.Context, req *ctlpb.ScanScmReq) (*c
 	return newScanScmResp(c.getScmUsage(ssr))
 }
 
+// Return the number of target managing a given NVME controleur
+func (c *ControlService) getTargetCount(pciAddr string) (int, error) {
+	for _, engineCfg := range c.srvCfg.Engines {
+		targetCount := engineCfg.TargetCount
+		for _, tierCfg := range engineCfg.Storage.Tiers {
+			if tierCfg.IsBdev() {
```
Comment (tanabarr) [inline src/control/server/ctl_storage_rpc.go:167]: nit: tend not to want to nest too deep, instead, break early. here use 'if !tierCfg.IsBdev() {continue' to break the loop and then you can de-indent the rest of the block. makes it nicer to read

File: src/control/server/ctl_storage_rpc.go [hunk 2]
```diff
@@ -151,6 +159,112 @@ func (c *ControlService) scanScm(ctx context.Context, req *ctlpb.ScanScmReq) (*c
...
+				for index := range tierCfg.Bdev.DeviceList {
+					if pciAddr == tierCfg.Bdev.DeviceList[index] {
+						msg := "NVME controler %s is used by %d targets"
+						c.log.Debugf(msg, pciAddr, targetCount)
```
Comment (tanabarr) [inline src/control/server/ctl_storage_rpc.go:171]: This doesn't make sense, targets will be split across different controllers (see `dmg storage scan --nvme-meta` command output) but from what you are doing here your messages indicate that the entire target count is used on each controller.

File: src/control/server/ctl_storage_rpc.go [hunk 3]
```diff
@@ -151,6 +159,112 @@ func (c *ControlService) scanScm(ctx context.Context, req *ctlpb.ScanScmReq) (*c
...
+						return targetCount, nil
+					}
+				}
+			}
+		}
+	}
+
+	return 0, errors.Errorf("unknown PCI address %s", pciAddr)
```
Comment (tanabarr) [inline src/control/server/ctl_storage_rpc.go:179]: the error is misleading if there are no engine configs

File: src/control/server/ctl_storage_rpc.go (hunk 2 above)
Comment (knard-intel) [reply src/control/server/ctl_storage_rpc.go:171]: From what I understand, I have a big issue on the algorithm itself.
I have to admit, that I did this part on empirical assumption (i.e. try and test).
I am going to look at the storage request to find the valid number of targets sharing a given nvme controler.
Thanks a lot for this feedback.

File: src/control/server/ctl_storage_rpc.go (hunk 3 above)
Comment (knard-intel) [reply src/control/server/ctl_storage_rpc.go:179]: Going to change it.

**Comment by knard-intel** [comment]
Close the PR until the issue on the number of targets per NVMe devices has been fixed.

**Pull request closed without merging**